spec/api/comment_thread_spec.rb

```diff
@@ -594,6 +594,24 @@ def thread_result(id, params)
         comment.endorsement.should == nil
         check_thread_result_json(nil, changed_thread, parse(last_response.body))
       end
+      it "update information of comment thread and mark thread as read" do
+        thread = CommentThread.first
+        comment = thread.comments.first
+        comment.endorsed = true
+        comment.endorsement = {:user_id => "42", :time => DateTime.now}
+        comment.save
+        put "/api/v1/threads/#{thread.id}", body: "new body", title: "new title", commentable_id: "new_commentable_id", thread_type: "question", read: true
+        last_response.should be_ok
+        changed_thread = CommentThread.find(thread.id)
+        changed_thread.body.should == "new body"
+        changed_thread.title.should == "new title"
+        changed_thread.commentable_id.should == "new_commentable_id"
+        changed_thread.thread_type.should == "question"
+        comment.reload
+        comment.endorsed.should == false
+        comment.endorsement.should == nil
+        check_thread_result_json(nil, changed_thread, parse(last_response.body))
```

Comment:
I agree with @cahrens.  This test seems the same as the one above it where we don't send read. 
* Does the first param of check_thread_result_json need to be the user?  Without this, it seems it is just checking for read to be false, but maybe I'm misreading the code.
* This test and the test above seem exactly the same except for the read param.
  * check_thread_result_json check seems to check that things are consistent, but not that they are accurate.  There should be a separate explicit assertion here for read to be true.
  * I'd change the earlier test to be named "update information of comment thread and don't mark thread as read" for consistency.  It should have an explicit assertion for read being false.